lib/event.c

```diff
@@ -601,13 +664,63 @@ struct event_loop *event_master_create(const char *name)
 	set_nonblocking(rv->io_pipe[0]);
 	set_nonblocking(rv->io_pipe[1]);
 
+#if EPOLL_ENABLED
+	rv->awakened = false;
+	pthread_mutex_init(&rv->awakened_mtx, NULL);
+
+	/* Initialize data structures for epoll() */
+	rv->handler.event_size = rv->fd_limit;
+	rv->handler.event_count = 0;
+	rv->handler.new_event_count = 0;
+	rv->handler.regular_revent_count = 0;
+	rv->handler.epoll_fd = epoll_create1(0);
+	if (rv->handler.epoll_fd == -1) {
+		flog_err(
+			EC_LIB_NO_THREAD,
+			"Couldn't create an epoll set!");
+		return NULL;
+	}
+	rv->handler.events =
+		XCALLOC(MTYPE_EVENT_MASTER, sizeof(struct frr_epoll_event) *
+						    rv->handler.event_size);
+	rv->handler.event_ptrs =
+		XCALLOC(MTYPE_EVENT_MASTER, sizeof(struct frr_epoll_event *) *
+						    rv->handler.event_size);
+	rv->handler.new_events =
+		XCALLOC(MTYPE_EVENT_MASTER, sizeof(struct frr_epoll_event) *
+						    rv->handler.event_size);
+	rv->handler.new_event_ptrs =
+		XCALLOC(MTYPE_EVENT_MASTER, sizeof(struct frr_epoll_event *) *
+						    rv->handler.event_size);
+	rv->handler.revents =
+		XCALLOC(MTYPE_EVENT_MASTER,
+			sizeof(struct epoll_event) * rv->handler.event_size);
+	rv->handler.regular_revents =
+		XCALLOC(MTYPE_EVENT_MASTER,
+			sizeof(struct epoll_event) * rv->handler.event_size);
+	for (i = 0; i < rv->handler.event_size; i++) {
+		rv->handler.events[i].index = -1;
+		rv->handler.new_events[i].index = -1;
+	}
+	pipe_read_ev.data.fd = rv->io_pipe[0];
+	pipe_read_ev.events = EPOLLIN;
+	if (-1 == epoll_ctl(rv->handler.epoll_fd, EPOLL_CTL_ADD, rv->io_pipe[0],
+			    &pipe_read_ev)) {
+		flog_err(
+			EC_LIB_NO_THREAD,
+			"Attempting to call epoll_ctl to add io_pipe[0] but failed, fd: %d!",
+			rv->io_pipe[0]);
+		return NULL;
```

Comment:
If we should be aborting the running process. Let's make this change.  FRR is dead in the water here as Quentin pointed out

Reply:
This is a fatal problem causing epoll to be inavailable, so we need aborting here.
I have changed the return statement to exit(1).